src/plugins/intel_cpu/src/utils/debug_capabilities.cpp

```diff
@@ -596,7 +658,250 @@ std::ostream & operator<<(std::ostream & os, const dnnl::memory::format_tag form
+                    os << sep << tag << opname[iop.get()] << "->output(" << out_port << ")";
+                } else {
+                    if (literal_consts.count(iop))
+                        os << sep << tag << literal_consts[iop];
+                    else
+                        os << sep << tag << opname[iop.get()];
+                }
+                sep = ", ";
+            }
+            // attributes are passed in using dict
+            os << "}, \"" << get_output_values_info(op) << "\"";
+
+            std::stringstream ss2;
+            OstreamAttributeVisitor osvis(ss2);
+            op->visit_attributes(osvis);
+            auto str_attr = ss2.str();
+            if (str_attr.size())
+                os << ", {" << str_attr << "}";
+            os << ");   //  " << op->get_friendly_name();
+
+            os << "(";
+            sep = "";
+            for (size_t i = 0; i < op->get_input_size(); i++) {
+                auto vout = op->get_input_source_output(i);
+                auto iop = vout.get_node_shared_ptr();
+                os << sep << tag << iop->get_friendly_name();
+                if (iop->get_output_size() > 1) {
+                    auto out_port = vout.get_index();
+                    os << "[" << out_port << "]";
+                }
+                sep = ", ";
+            }
+            os << ")" << std::endl;;
+
+            //auto op_output_size = op->get_output_size();
+            //if (op_output_size > 1) {
+            //    os << "    " << name << "->set_output_size(" << op_output_size << ");" << std::endl;
+            //}
+        }
+
+        // recursively output subgraphs
+        if (auto msubgraph = std::dynamic_pointer_cast<op::util::MultiSubGraphOp>(op)) {
+            auto cnt = msubgraph->get_internal_subgraphs_size();
+            for (size_t i = 0; i < cnt; i++) {
+                os << "    MultiSubGraphOp " << tag << msubgraph->get_friendly_name() << "[" << i << "]" << std::endl;
+                dump_cpp_style(os, msubgraph->get_function(i));
+            }
+        }
+    }
+    os << prefix << "}\n";
+}
+
+DumpModel::DumpModel(std::string _file_name) {
+    static int dump_index = 0;
+    static const char* exec_graph_path = std::getenv("OV_CPU_EXEC_GRAPH_PATH");
+    if (!exec_graph_path)
+        return;
+
+    std::string path(exec_graph_path);
+    if (path.size() >= 4 && path.compare(path.size() - 4, 4, ".cpp"))
+        return;
+
+    file_name = "dump_model" + std::to_string(dump_index) + "_" + _file_name;
+    dump_index++;
+}
+
+bool DumpModel::run_on_model(const std::shared_ptr<ov::Model>& model) {
+    if (file_name.empty())
+        return false;
+
+    std::ofstream ofs(file_name);
+    if (!ofs) {
+        std::cout << "Error opening file " << file_name << " for output" << std::endl;
+        return false;
+    }
+    dump_cpp_style(ofs, model);
+    ofs.close();
+
+    ov::serialize(model, file_name + ".xml", "/dev/null");
```

Comment:
I think we should not silently create an xml file as well.
OV_CPU_DUMP_IR supports specifying multiple formats.